lib/rubygems/precompiler.rb

```diff
@@ -87,6 +107,18 @@ def build_extensions(install_root)
       tempdir do |workroot|
         extract_files_into(workroot)
 
+        # override the full_gem_path function so we can return
+        # the directory we want. Otherwise by default the build process will
+        # look for the gem installed in the usual place won't find it and will then
+        # bail
+        class <<@spec
+          attr_accessor :workroot
+          def full_gem_path
+            return workroot
+          end
+        end
+        @spec.workroot = workroot
```

Comment:
Possible but we also add an attr_accessor, and this version of the code is what is already deployed and tested in production. (It just never got merged to master but we build off a sha). In the interest of knowing it already works and limiting the number of changes id leave it as-is for now. Happy too look at refactoring in a separate release though.